api/tasks.go

```diff
@@ -595,11 +597,29 @@ type DispatchPayloadConfig struct {
 	File string
 }
 
+type TaskLifecycle struct {
+	Hook       string        `mapstructure:"hook"`
+	BlockUntil string        `mapstructure:"block_until"`
+	Deadline   time.Duration `mapstructure:"deadline"`
+}
+
+func (l *TaskLifecycle) Canonicalize() {
+	if l.Deadline == 0 {
+		l.Deadline = structs.TaskLifecycleDeadlineDefault
```

Comment:
Copy this value in api package to avoid having api depending on internal nomad/structs package